model_compression_toolkit/data_generation/common/enums.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2023 Sony Semiconductor Israel, Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+from enum import Enum
+
+
+class EnumBaseClass(Enum):
+    @classmethod
+    def get_values(cls):
+        """
+        Get the list of values corresponding to the enum members.
+
+        Returns:
+            List of values.
+        """
+        return [mem.value for mem, mem in cls.__members__.items()]
+
+
+class ImageGranularity(EnumBaseClass):
+    """
+    An enum for choosing the image dependence granularity when generating images.
+    0. ImageWise
+    1. BatchWise
+    2. AllImages
+    """
+
+    ImageWise = 0
+    BatchWise = 1
+    AllImages = 2
+
+
+class DataInitType(EnumBaseClass):
+    """
+    An enum for choosing the image dependence granularity when generating images.
+    0. Gaussian
+    1. Diverse
+    """
+    Gaussian = 0
+    Diverse = 1
+
+
+class ImagePipelineType(EnumBaseClass):
+    """
+    An enum for choosing the image pipeline type for image manipulation.
+    CROP_FLIP: Crop and flip the images.
+    IDENTITY: Do not apply any manipulation (identity transformation).
+    """
+    CROP_FLIP = 'crop_flip'
+    IDENTITY = 'identity'
```

Comment:
Because I will need to add "if" statements in a couple of places in the code only for the "identity" case, furthermore, the default is "crop flip" so usually this is not used.